File: dataset/human/c/file_9684.c
/*自由木を作成し、それらの情報を書き出すプログラム*/
#include<stdio.h>

/*マクロ*/
#define MAX 100005
#define NIL -1

/*グローバル変数*/
int n, D[MAX];//n=接点の個数
int dep = 0;

/*プロトタイプ宣言*/
void TreePrint(int );

/*構造体(入れ子)*/
struct Node{ 
  int p, l, r;
};

struct Node T[MAX];



int main(){
  int i, j;//カウンター変数
  int d, v, c, l;//d=degree, v=verteck, c=child, l=leaf`
  int u = 0;
  int dep = 0;

  scanf("%d", &n);

  for(i = 0; i < n; i++) {
    T[i].p = NIL;
    T[i].l = NIL;
    T[i].r = NIL;
  }
  
  for(i = 0; i < n; i++) {
    scanf("%d %d", &v, &d);

    for(j = 0; j < d; j++) {
      scanf("%d", &c);

      T[c].p = v;

      if(j == 0) {
	T[v].l = c;
      }

      if(j > 0) {
	T[l].r = c;
      }
	l = c;
    }

  }


  for(i = 0 ;i < n ;i++) {
    u = i;

  while(1)
    {
    if(T[u].p == NIL)  {
      break;
    }
    // printf("%d\n",dep);
    dep++;
    u = T[u].p;
    }

    D[i] = dep;
    dep = 0;
  }
  
  for(i = 0 ;i < n ;i++) {

    TreePrint(i);
  }
 
  return 0;
}

/*結果を出力する関数*/
void TreePrint(int u) {
  int i;

  printf("node %d: parent = %d, depth = %d, ",u ,T[u].p ,D[u]);

  if(T[u].p == NIL) {
    printf("root, ");
  }

  else if(T[u].l == NIL) {
    printf("leaf, ");
  }

  else if(T[u].l != NIL) {
    printf("internal node, ");
  }

  printf("[");

  if(T[u].l != NIL) {
    printf("%d",T[u].l);
    u=T[u].l;
  

    if(T[u].r != NIL)printf(", ");

    while(1){
	if(T[u].r == NIL)break;

      printf("%d",T[u].r);
      u=T[u].r;

      if(T[u].r != NIL) {
	printf(", ");
      }

    }
  }
  
  printf("]\n");
}

File: dataset/human/java/file_9684.java
import java.util.Scanner;

class Node {
	int i, id, left, right, parent = -1, sibling = -1, degree = 0, depth = 0, height = 0;

	public Node(int Id, int l, int r) {
		id = Id;
		left = l;
		right = r;
	}

	public void Show() {

		System.out.print("node " + id + ": parent = " + parent + ", sibling = " + sibling + ", degree = " + degree
				+ ", depth = " + depth + ", height = " + height + ", ");
		if (parent == -1) {
			System.out.println("root");
		} else if (degree == 0) {
			System.out.println("leaf");
		} else {
			System.out.println("internal node");
		}
	}
}

public class Main {
	static Node[] node = new Node[100000];

	public static void hight(int i) {
		if (node[i].parent != -1 && node[node[i].parent].height < node[i].height + 1) {
			node[node[i].parent].height = node[i].height + 1;
			hight(node[i].parent);
		}
	}

	public static void depth(int i) {
		if (node[i].left != -1) {
			node[node[i].left].depth=node[i].depth+1;
			depth(node[i].left);
		}
		if (node[i].right != -1) {
			node[node[i].right].depth=node[i].depth+1;
			depth(node[i].right);
		}
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int i, n, id, l, r;
		n = scan.nextInt();
		for (i = 0; i < n; i++) {
			id = scan.nextInt();
			l = scan.nextInt();
			r = scan.nextInt();
			node[id] = new Node(id, l, r);
		}
		for (i = 0; i < n; i++) {
			if (node[i].left != -1) {
				node[i].degree++;
				node[node[i].left].parent = node[i].id;
				node[node[i].left].sibling = node[i].right;
			}
			if (node[i].right != -1) {
				node[i].degree++;
				node[node[i].right].parent = node[i].id;
				node[node[i].right].sibling = node[i].left;
			}
		}
		for (i = 0; i < n; i++) {
			if (node[i].parent == -1) {
				depth(i);
			}
			if (node[i].degree == 0) {
				hight(i);
			}
		}
		for (i = 0; i < n; i++) {
			node[i].Show();
		}
		scan.close();
	}
}
